src/main/java/sirius/web/controller/BasicController.java

```diff
@@ -161,15 +163,72 @@ public void fail(WebContext ctx, HandledException error) {
     }
 
     @Override
-    public void onError(WebContext ctx, HandledException error) {
-        if (ctx.isResponseCommitted() || defaultRoute == null) {
-            fail(ctx, error);
+    public void onError(WebContext webContext, HandledException error) {
+        if (webContext.isResponseCommitted() || defaultRoute == null) {
+            fail(webContext, error);
             return;
         }
 
         if (error != null) {
             UserContext.message(Message.error(error.getMessage()));
         }
-        defaultRoute.accept(ctx);
+        defaultRoute.accept(webContext);
+    }
+
+    @Override
+    public void onJsonError(WebContext webContext, HandledException error) {
+        if (webContext.isResponseCommitted()) {
+            // Force underlying request / response to be closed...
+            webContext.respondWith()
+                      .error(HttpResponseStatus.INTERNAL_SERVER_ERROR, Exceptions.handle(ControllerDispatcher.LOG, error));
+            return;
+        }
+
+        if (error instanceof ErrorCodeException) {
+            try {
+                createJsonResult(webContext,
+                                 error,
+                                 HttpResponseStatus.valueOf(Integer.parseInt(((ErrorCodeException) error).getCode())));
```

Comment:
no add status as extra field in ErrorCodeException - the code field seems to be s.th. different

Reply:
added HttpResponseStatus to ErrorCodeException